packages/react/src/components/ComposedModal/ComposedModal.tsx

```diff
@@ -0,0 +1,457 @@
+        onMouseDown={handleMousedown}
+        onKeyDown={handleKeyDown}
+        className={modalClass}>
+        <div
+          className={containerClass}
+          role="dialog"
+          aria-modal="true"
+          aria-label={ariaLabel ? ariaLabel : generatedAriaLabel}
+          aria-labelledby={ariaLabelledBy}>
+          {/* Non-translatable: Focus-wrap code makes this `<button>` not actually read by screen readers */}
+          <button
+            type="button"
+            ref={startSentinel}
+            className={`${prefix}--visually-hidden`}>
+            Focus sentinel
+          </button>
+          <div ref={innerModal} className={`${prefix}--modal-container-body`}>
+            {childrenWithProps}
+          </div>
+          {/* Non-translatable: Focus-wrap code makes this `<button>` not actually read by screen readers */}
+          <button
+            type="button"
+            ref={endSentinel}
+            className={`${prefix}--visually-hidden`}>
+            Focus sentinel
+          </button>
+        </div>
+      </div>
+    );
+  }
+);
+
+ComposedModal.propTypes = {
+  /**
+   * Specify the aria-label for bx--modal-container
+   */
+  ['aria-label']: PropTypes.string,
+
+  /**
+   * Specify the aria-labelledby for bx--modal-container
+   */
+  ['aria-labelledby']: PropTypes.string,
+
+  /**
+   * Specify the content to be placed in the ComposedModal
+   */
+  children: PropTypes.node,
+
+  /**
+   * Specify an optional className to be applied to the modal root node
+   */
+  className: PropTypes.string,
+
+  /**
+   * Specify an optional className to be applied to the modal node
+   */
+  containerClassName: PropTypes.string,
+
+  /**
+   * Specify whether the primary button should be replaced with danger button.
+   * Note that this prop is not applied if you render primary/danger button by yourself
+   */
+  danger: PropTypes.bool,
+
+  /**
+   * Specify whether the Modal content should have any inner padding.
+   */
+  isFullWidth: PropTypes.bool,
+
+  /**
+   * Specify an optional handler for closing modal.
+   * Returning `false` here prevents closing modal.
+   */
+  onClose: PropTypes.func,
+
+  /**
+   * Specify an optional handler for the `onKeyDown` event. Called for all
+   * `onKeyDown` events that do not close the modal
+   */
+  onKeyDown: PropTypes.func,
+
+  /**
+   * Specify whether the Modal is currently open
+   */
+  open: PropTypes.bool,
+
+  preventCloseOnClickOutside: PropTypes.bool,
+
+  /**
+   * Specify a CSS selector that matches the DOM element that should be
+   * focused when the Modal opens
+   */
+  selectorPrimaryFocus: PropTypes.string,
+
+  /**
+   * Specify the CSS selectors that match the floating menus
+   */
+  selectorsFloatingMenus: PropTypes.arrayOf(PropTypes.string),
+
+  /**
+   * Specify the size variant.
+   */
+  size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg']),
+};
+ComposedModal.defaultProps = {
+  selectorPrimaryFocus: '[data-modal-primary-focus]',
+};
```

Comment:
what happened to the onKeyDown default prop here? 
<img width="483" alt="image" src="https://user-images.githubusercontent.com/40550942/233739922-2251412c-1715-4506-be4f-4e02fa15a9be.png">

Reply:
It is much more readable and makes more sense to skip the defaulting to no-op and either condition the consumer site or add the fallback in desctructuring... there are occurrences of both of these patterns elsewhere in the codebase, so I've opted for it here as well.

Also, one aspect that helps here specifically is that plays much nicer with typescript because if you supply defaultProps for a property, the pops entry must be non-nullable

Reply:
Regarding the failing test ... as far as I'm aware, it's about typescript errors around the invalid html attribute